### ISSUE TITLE
Adapter doesn't play nicely with validation rules

### ISSUE THREAD

**Issue by courthead** [firebase/emberfire#304]
Firebase allows you to have validation rules ensuring that particular children exist for any given node. For example:

```
{
  "rules" : {
    "posts" : {
      "$postsID" : {
        ".validate" : "newData.hasChildren(['title','body','authors'])",
        ...
```

Above, if we try to create a node at `/posts/<post-id>` without giving it a title, body, or authors, Firebase will reject our request.

However, the `updateRecord` function in the [EmberFire adapter](https://github.com/firebase/emberfire/blob/master/addon/adapters/firebase.js) is using separate promises to save relationships and the model itself.

So, in the example above, when we save a new post, EmberFire might attempt to write to `/posts/<post-id>/authors/<author-id>` before it writes to `/posts/<post-id>/title` and `/posts/<post-id>/body`. In that case, our validation rules will fail.

I believe the solution is to write all the updates to a node using a single request instead of using multiple requests.

**Comment by tstirrat** [comment]
It's definitely something I can look into. To be honest, I'm not 100% sure why relationship updates are done in separate calls, this piece was done way before my time. I assume it was to cover for some realtime collaboration edge case.

I will try to discover the original reason behind it. If I can't find out a good reason why it was done this way I am happy to update the adapter to save with a single call.

@aputinski any thoughts on why it was originally done this way?

**Comment by adjohnson916** [comment]
+1

**Comment by jamesdaniels** [comment]
This will be fixed with #366
